Composer/packages/extensions/adaptive-form/src/utils/uiOptionsHelpers.ts

```diff
@@ -39,15 +39,22 @@ export function getUiPlaceholder(props: FieldProps): string | undefined {
 
   const { placeholder: uiPlaceholder } = uiOptions;
 
+  let fieldUIPlaceholder = '';
+
   if (uiPlaceholder) {
-    return typeof uiPlaceholder === 'function' ? uiPlaceholder(value) : uiPlaceholder;
+    fieldUIPlaceholder = typeof uiPlaceholder === 'function' ? uiPlaceholder(value) : uiPlaceholder;
+  } else if (placeholder) {
+    fieldUIPlaceholder = placeholder;
+  } else if (schema && Array.isArray(schema.examples) && schema.examples.length > 0) {
+    fieldUIPlaceholder = `ex. ${schema.examples.join(', ')}`;
   }
 
-  if (placeholder) {
-    return placeholder;
+  if (fieldUIPlaceholder && schema.pattern) {
+    const regex = `${schema.pattern}`;
+    const placeholderExamples = fieldUIPlaceholder.split(',').map((example) => example.trim());
+    const filteredExamples = placeholderExamples.filter((example) => example.match(regex));
+    fieldUIPlaceholder = filteredExamples.join(', ');
   }
 
-  if (schema && Array.isArray(schema.examples) && schema.examples.length > 0) {
-    return `ex. ${schema.examples.join(', ')}`;
-  }
+  return fieldUIPlaceholder !== '' ? fieldUIPlaceholder : undefined;
```

Comment:
Can you show me an example schema that you were thinking about when coding this? I don't see how this fixes the problem yet.

Reply:
The screenshot from the PR description has a good example. You can see there than the schema has a regex pattern for strings vs expressions and that filtering the examples with those patterns greatly help getting more refined examples.

Reply:
Ah ok. So this uses the $ref's pattern to match against the "parent's" examples? That makes more sense.